docs/new-architecture-backward-turbomodule.md

```diff
@@ -0,0 +1,709 @@
+
+We already run some of the steps to connect the module with the app.
+
+Open the `NewArchitectureApp/App.js` and replace the code with the same one we used in the `OldArchitectureApp`.
+
+```js
+/**
+ * @format
+ * @flow strict-local
+ */
+
+import React from 'react';
+import { useState, useEffect } from 'react';
+import type { Node } from 'react';
+import {
+  SafeAreaView,
+  StatusBar,
+  Text,
+  Button
+} from 'react-native';
+
+import type { AuthorizationStatus } from 'example-library/src/index';
+import AuthorizationManager from 'example-library/src/index';
+
+const App: () => Node = () => {
+  const [authorization, setAuthorization] =
+    useState<AuthorizationStatus>('NotDetermined');
+
+  useEffect(() => {
+    async function setInitialState() {
+      const newState =
+        await AuthorizationManager.getCurrentStatus();
+      setAuthorization(newState);
+    }
+    setInitialState();
+  }, []);
+
+  return (
+    <SafeAreaView style={backgroundStyle}>
+      <StatusBar
+        barStyle={isDarkMode ? 'light-content' : 'dark-content'}
+      />
+      <Button
+        title="Request Authorization"
+        onPress={() => {
+          AuthorizationManager.askAuthorization();
+        }}
+      />
+      <Text>Current authorization is: {authorization}</Text>
+    </SafeAreaView>
+  );
+};
+
+export default App;
+```
+
+Now, the only things we have to change are the `import` statements. Replace the `import type { AuthorizationStatus }` and the `import AuthorizationManager` lines with the following:
+
+```js
+import type { AuthorizationStatus } from 'example-library/src/AuthorizationStatus';
+import AuthorizationManager from 'example-library/src/NativeAuthorizationManager';
+```
+
+Before running the app, remember to add the required `Info.plist` keys:
+
+- `NSLocationWhenInUseUsageDescription`
+- `NSLocationAlwaysAndWhenInUseUsageDescription`
+
+Now, from the terminal, run the `npx react-native run-ios` command to see the app working.
+
+## Summary
+
+To summarise, the steps to create a Turbomodule which is bacward compatible with the old architecture, are:
+
+1. Create a Native Module.
+2. Create the Turbomodule specs.
+3. Make sure that the interfaces of the Native Module and of the Turbomodule are the same.
+4. Update the module podspec, using the `ENV["RCT_NEW_ARCH_ENABLED"] == "1"` guard to avoid installing the dependencies in the old architecture
+5. Add the `codegenConfig` to the `package.json` of the module
+6. Add the `#import "<moduleName>Spec.h`, guarded by an `#ifdef RCT_NEW_ARCH_ENABLED` guard
+7. Implement the `getTurboModule` C++ method in the native code, guarded by an `#ifdef RCT_NEW_ARCH_ENABLED` guard.
+
+Then, in your app, you can choose which version of the module to use with the proper `import` in JS and installing the pods with the right command:
+
+- `pod install` for the old architecture
+- `RCT_NEW_ARCH_ENABLED=1 pod install` for the new one
```

Comment:
Probably we want add note for M1 users in there. There is a partial for that named `_markdown-m1-cocoapods.mdx`.